driver/k8s/logicalvolume_service.go

```diff
@@ -221,6 +224,53 @@ func (s *LogicalVolumeService) DeleteVolume(ctx context.Context, volumeID string
 	}
 }
 
+// CreateSnapshot creates a snapshot of existing volume.
+func (s *LogicalVolumeService) CreateSnapshot(ctx context.Context, node, dc, sourceVolID, sname, snapType, accessType string, sourceVol *topolvmv1.LogicalVolume) (string, error) {
+	logger.Info("CreateSnapshot called", "name", sname)
+	snapshotLV := &topolvmv1.LogicalVolume{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "LogicalVolume",
+			APIVersion: "topolvm.cybozu.com/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: sname,
+		},
+		Spec: topolvmv1.LogicalVolumeSpec{
+			Name:        sname,
+			NodeName:    node,
+			DeviceClass: dc,
+			Size:        sourceVol.Spec.Size,
+			Type:        snapType,
+			SourceID:    sourceVol.Status.VolumeID,
```

Comment:
sourceVolID is an argument to the function. Can't that be used instead?
Should the function take the size as an argument instead of getting it from the Sourcevol which is not required otherwise?

Reply:
Sounds good. We can get rid of `sourceVol` parameter as a whole